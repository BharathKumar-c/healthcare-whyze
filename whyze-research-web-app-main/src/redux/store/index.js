import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
