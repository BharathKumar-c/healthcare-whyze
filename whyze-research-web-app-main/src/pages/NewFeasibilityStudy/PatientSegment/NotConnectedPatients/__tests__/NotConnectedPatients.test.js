import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import React from 'react';
import { NotConnectedPatients } from '..';

function renderWithContext(element) {
  const { container, getByText, getByRole, getByLabelText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText, getByRole, getByLabelText };
}

describe('create tests for NotConnectedPatients', () => {
  test('should create snapshots', () => {
    const { container } = renderWithContext(<NotConnectedPatients />);
    expect(container).toMatchSnapshot();
  });
  test('should check title render correctly', () => {
    const { container } = renderWithContext(<NotConnectedPatients />);
    const siteStext = container.getElementsByClassName('sites-text');

    expect(siteStext[0].firstChild.textContent).toBe('Not connected to sites');
  });

  test('should check NotConnectedPatients render correctly', () => {
    const { container } = renderWithContext(
      <NotConnectedPatients noOfNotConncetedPatients={1000} />,
    );
    expect(
      container.getElementsByClassName('no-of-patients-text')[0].firstChild
        .textContent,
    ).toBe('1000');
  });
});
