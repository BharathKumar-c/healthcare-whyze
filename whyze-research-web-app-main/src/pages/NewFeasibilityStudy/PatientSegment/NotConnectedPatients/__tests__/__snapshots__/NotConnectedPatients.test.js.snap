// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create tests for NotConnectedPatients should create snapshots 1`] = `
<div>
  <div
    class="not-connected-patients-container"
  >
    <div
      class="alignment"
    >
      <svg>
        UserDelete.svg
      </svg>
      <div
        class="no-of-patients-text"
      >
        0
      </div>
    </div>
    <div
      class="alignment"
    >
      <div
        class="sites-text"
      >
        Not connected to sites
      </div>
      <span
        aria-label="info-circle"
        class="anticon anticon-info-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="info-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
